app/controllers/cohort_imports_controller.rb

```diff
@@ -28,25 +27,43 @@ def create
 
     @cohort_import.process!
 
-    @cohort_import.record!
+    if @cohort_import.processed_only_exact_duplicates?
+      render :duplicates
+      return
+    end
 
-    redirect_to action: :success
+    redirect_to edit_programme_cohort_import_path(@programme, @cohort_import)
   end
 
-  def success
+  def show
+  end
+
+  def edit
+    render layout: "full"
+  end
+
+  def update
+    @cohort_import.record!
+
+    redirect_to programme_cohort_import_path(@programme, @cohort_import)
   end
 
   private
 
-  def cohort_import_params
-    params.fetch(:cohort_import, {}).permit(:csv)
+  def set_programme
+    @programme = policy_scope(Programme).active.find(params[:programme_id])
   end
 
-  def set_team
-    @team = current_user.team
+  def set_cohort_import
+    # TODO: @programme.cohort_imports.find
```

Comment:
Follow-up?

Reply:
Yeah, although I'm going to wait until we've settled on the correct data model for cohorts. I think there will be a relationship between cohort imports and programmes, but not entirely sure yet.